pkg/sdk/model/output/http.go

```diff
@@ -0,0 +1,147 @@
+	// Read timeout in seconds.
+	ReadTimeout int `json:"read_timeout,omitempty"`
+	// TLS timeout in seconds.
+	SSLTimeout int `json:"ssl_timeout,omitempty"`
+	// The default version of TLS transport. [TLSv1_1, TLSv1_2] (default: TLSv1_2)
+	TlsVersion string `json:"tls_version,omitempty"`
+	// The cipher configuration of TLS transport. (default: ALL:!aNULL:!eNULL:!SSLv2)
+	TlsCiphers string `json:"tls_ciphers,omitempty"`
+	// The CA certificate path for TLS.
+	TlsCACertPath *secret.Secret `json:"tls_ca_cert_path,omitempty"`
+	// The client certificate path for TLS.
+	TlsClientCertPath *secret.Secret `json:"tls_client_cert_path,omitempty"`
+	// The client private key path for TLS.
+	TlsPrivateKeyPath *secret.Secret `json:"tls_private_key_path,omitempty"`
+	// The client private key passphrase for TLS.
+	TlsPrivateKeyPassphrase *secret.Secret `json:"tls_private_key_passphrase,omitempty"`
+	// The verify mode of TLS. [peer, none] (default: peer)
+	TlsVerifyMode string `json:"tls_verify_mode,omitempty"`
+	// Raise UnrecoverableError when the response code is non success, 1xx/3xx/4xx/5xx. If false, the plugin logs error message instead of raising UnrecoverableError. (default: true)
+	ErrorResponseAsUnrecoverable bool `json:"error_response_as_unrecoverable,omitempty"`
+	// List of retryable response codes. If the response code is included in this list, the plugin retries the buffer flush. (default: [503])
+	RetryableResponseCodes []int `json:"retryable_response_codes,omitempty"`
+	// +docLink:"HTTP auth,#http_auth"
+	Auth *HTTPAuth `json:"auth,omitempty"`
+	// +docLink:"Buffer,../buffer/"
+	Buffer *Buffer `json:"buffer,omitempty"`
+}
+
+func (c *HTTPOutputConfig) ToDirective(secretLoader secret.SecretLoader, id string) (types.Directive, error) {
+	pluginType := "http"
+	http := &types.OutputPlugin{
+		PluginMeta: types.PluginMeta{
+			Type:      pluginType,
+			Directive: "match",
+			Tag:       "**",
+			Id:        id,
+		},
+	}
+	if params, err := types.NewStructToStringMapper(secretLoader).StringsMap(c); err != nil {
+		return nil, err
+	} else {
+		http.Params = params
+	}
+	if c.Buffer != nil {
+		if buffer, err := c.Buffer.ToDirective(secretLoader, id); err != nil {
+			return nil, err
+		} else {
+			http.SubDirectives = append(http.SubDirectives, buffer)
+		}
+	}
+	if c.Format != nil {
+		if format, err := c.Format.ToDirective(secretLoader, ""); err != nil {
+			return nil, err
+		} else {
+			http.SubDirectives = append(http.SubDirectives, format)
+		}
+	}
+	if c.Auth != nil {
+		if auth, err := c.Auth.ToDirective(secretLoader, ""); err != nil {
+			return nil, err
+		} else {
+			http.SubDirectives = append(http.SubDirectives, auth)
+		}
+	}
+	return http, nil
+}
+
+// +kubebuilder:object:generate=true
+// +docName:"HTTP auth config"
+// http_auth
+type HTTPAuth struct {
+	// Username for basic authentication.
+	Username string `json:"username"`
+	// Password for basic authentication.
+	Password string `json:"password"`
```

Comment:
This should be a `secret.Secret` I beleive.

Reply:
done. notice that these fields are actually required, but the current implementation doesn't recursively handle tags like `plugin:"required"` for struct field and I didn't want to dive too deep into a major refactor.